samtranslator/feature_toggle/feature_toggle.py

```diff
@@ -0,0 +1,126 @@
+        :param region: region in which SAM is running
+        :return:
+        """
+        if feature_name not in self.feature_config:
+            LOG.warning("Feature '{}' not available in Feature Toggle Config.".format(feature_name))
+            return False
+        stage_config = self.feature_config.get(feature_name, {}).get(stage, {})
+        if not stage_config:
+            LOG.info("Stage '{}' not enabled for Feature '{}'.".format(stage, feature_name))
+            return False
+        account_config = stage_config.get(account_id) if account_id in stage_config else stage_config.get("default", {})
+        region_config = (
+            account_config.get(region, {}) if region in account_config else account_config.get("default", {})
+        )
+        is_enabled = region_config.get("enabled", False)
+        LOG.info("Feature '{}' is enabled: '{}'".format(feature_name, is_enabled))
+        return is_enabled
+
+
+class FeatureToggleConfigProvider:
+    """Interface for all FeatureToggle config providers"""
+
+    def __init__(self):
+        pass
+
+    @property
+    def config(self):
+        raise NotImplementedError
+
+
+class FeatureToggleDefaultConfigProvider(FeatureToggleConfigProvider):
+    """Default config provider, always return False for every query."""
+
+    def __init__(self):
+        FeatureToggleConfigProvider.__init__(self)
+
+    @property
+    def config(self):
+        return {}
+
+
+class FeatureToggleLocalConfigProvider(FeatureToggleConfigProvider):
+    """Feature toggle config provider which uses a local file. This is to facilitate local testing."""
+
+    def __init__(self, local_config_path=os.path.join(my_path, "..", "..", "bin", "feature_toggle_config.json")):
+        FeatureToggleConfigProvider.__init__(self)
+        with open(local_config_path, "r") as f:
+            config_json = f.read()
+        self.feature_toggle_config = json.loads(config_json)
+
+    @property
+    def config(self):
+        return self.feature_toggle_config
+
+
+class FeatureToggleAppConfigConfigProvider(FeatureToggleConfigProvider):
+    """Feature toggle config provider which loads config from AppConfig."""
+
+    def __init__(self, application_id, environment_id, configuration_profile_id):
+        FeatureToggleConfigProvider.__init__(self)
+        self.app_config_client = boto3.client("appconfig")
+        try:
+            response = self.app_config_client.get_configuration(
+                Application=application_id,
+                Environment=environment_id,
+                Configuration=configuration_profile_id,
+                ClientId="FeatureToggleAppConfigConfigProvider",
+            )
+            binary_config_string = response["Content"].read()
+            self.feature_toggle_config = json.loads(binary_config_string.decode("utf-8"))
+        except Exception as ex:
+            LOG.error("Failed to load config from AppConfig: {}".format(ex))
+            LOG.info("Falling to empty config.")
```

Comment:
Can probably just remove this - the info line adds no debugging value since there's no logical fork here. If needed just add this snippet to the error line.

Reply:
Done